pkg/api/vm/handler.go

```diff
@@ -0,0 +1,61 @@
+package vm
+
+import (
+	"net/http"
+
+	ctlkubevirtv1alpha3 "github.com/rancher/vm/pkg/generated/controllers/kubevirt.io/v1alpha3"
+
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type vmActionHandler struct {
+	vms ctlkubevirtv1alpha3.VirtualMachineClient
+}
+
+func (h *vmActionHandler) ServeHTTP(rw http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	action := vars["action"]
+	namespace := vars["namespace"]
+	name := vars["name"]
+
+	running := false
+	switch action {
```

Comment:
`restart` VM action is also required, and only when the VM running spec is not equal to false, refer to https://kubevirt.io/user-guide/#/architecture?id=restarting.